### ISSUE TITLE
I can't compile any code using the api.

### ISSUE THREAD

**Issue by xXDarkCubeXx** [FluidSynth/fluidsynth#925]
### Describe the bug
I can't compile code using the api.

### Expected behavior
I should be able to compile code using the api?

### Steps to reproduce
this is the header file:
``` C
#ifndef GETAUDIOSAMPLES_H_INCLUDED
#define GETAUDIOSAMPLES_H_INCLUDED
#include </fluidsynth/include/fluidsynth.h>

float *buff[2];
fluid_synth_t* synth;
fluid_settings_t* settings;
fluid_player_t* player;
void setup_synth(char* midi_file, char* sfs[], int nsfs);
void fill_buffers(int n_samples);

#endif // GETAUDIOSAMPLES_H_INCLUDED
```
and the main file is this:
``` C
#include "GetAudioSamples.h"
#include <assert.h>
#include <stdio.h>
#include <string.h>

void setup_synth(char* midi_file, char* sfs[], int nsfs){
    settings = new_fluid_settings();
    synth = new_fluid_synth(settings);
    player = new_fluid_player(synth);
    assert(fluid_is_midifile(midi_file));
    fluid_player_add(player, midi_file);
    for (int i=0;i<nsfs;i++){
        fluid_synth_sfload(synth, sfs[i], 0);
        fluid_synth_program_select(synth, i, i, 0, 0);
    }
}

void fill_buffers(int n_samples){
    float left[n_samples], right[n_samples];
    // array of buffers used to setup channel mapping
    // first make sure to zero out the sample buffers everytime before calling fluid_synth_process()
    memset(left, 0, sizeof(left));
    memset(right, 0, sizeof(right));
    // setup channel mapping for a single stereo channel to which to render all dry audio to
    buff[0] = left;
    buff[1] = right;
    int err = fluid_synth_process(synth, n_samples, 0, NULL, 2, buff);
    if(err==FLUID_FAILED){
        puts("oops");
    }
}

int main()
{
    char *sfs[1] = {"C:/Users/amran/Music/sf/piano"};
    setup_synth("C:/Users/amran/test.mid", sfs, 1);
    fill_buffers(512);
    printf("%d", buff[0][0]);
    return 0;
}
```

I'm on windows and I got fluidsynth from the latest release.

### Additional context
Here is the build message:
```
||=== Build: default in untitled (compiler: GNU GCC Compiler) ===|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__new_fluid_settings'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__new_fluid_synth'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__new_fluid_player'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__fluid_is_midifile'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__fluid_player_add'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__fluid_synth_sfload'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__fluid_synth_program_select'|
c:\mingw\bin\..\lib\gcc\mingw32\9.2.0\..\..\..\..\mingw32\bin\ld.exe: .objs\GetAudioSamples.o:GetAudioSamples.c|| undefined reference to `_imp__fluid_synth_process'|
||error: ld returned 1 exit status|
||=== Build failed: 9 error(s), 0 warning(s) (0 minute(s), 3 second(s)) ===|
```

I opened a discussion but I didn't get any useful answers in 2 days so I'm reposting it here. I hope that's fine.

**Comment by derselbst** [comment]
> I opened a discussion but I didn't get any useful answers in 2 days so I'm reposting it here. I hope that's fine.

Wow, 2 days. Really. This level of patience deserves an immediate answer... next week.

**Comment by xXDarkCubeXx** [comment]
wut? I'm just asking for help man.

**Comment by derselbst** [comment]
> I'm just asking for help man

But not by filing bug tickets.

> is it something in my code?

It's your fault, yes. It's not obvious because you haven't provided that particular information I'm looking for. But my magic crystal ball is very confident. Also, I've tested it myself and  can reproduce it. And if you hadn't been so impatient and crude asking the same question again by filing a bug ticket, even though it clearly says in capital letters that bugs should not be abused for questions, I would have already given you the answer. In the meantime, you can fix the typo in your Github biography.

**Comment by xXDarkCubeXx** [comment]
> In the meantime, you can fix the typo in your Github biography.

you must've felt so witty writing that lol. Dude, I was 15 when I wrote that bio and english isn't even my first language.

> even though it clearly says in capital letters that bugs should not be abused for questions

I wasn't sure if it was a bug or not, bro even you thought it was a bug. remember the issue you linked?

> And if you hadn't been so impatient and crude... I would have already given you the answer.

Look man, I'd love to fix this as fast as possible yes, but ofc I'm gonna be impatient when it takes you a day to reply every time! you just don't want to help me, which is understandable, you can do whatever you want I guess. 

You're really being a dick to me and I don't know why. Although, it seems you have contributed quite a lot to this repository (biggest contributor of all time by lines changed). Fluidsynth is helping me a lot so thank you for that, I really appreciate it.
if you don't want to keep maintaining this open source project just stop man. no one is forcing you to do it.

**Comment by mawe42** [comment]
> if you don't want to keep maintaining this open source project just stop man. no one is forcing you to do it.

That attitude might be the reason why @derselbst reacted so strongly to you opening this issue. You seem to be under the impression that we are obliged to solve your problem. And that we should solve it fast. The statement "no useful answers in 2 days", when derselbst reacted to your question 12 hours after you posted it is quite offensive. Your statement "ofc I'm gonna be impatient when it takes you a day to reply every time!" is also quite telling.

Let me suggest some reading material, it might be enlightening: https://snarky.ca/the-social-contract-of-open-source/

**Comment by xXDarkCubeXx** [comment]
> You seem to be under the impression that we are obliged to solve your problem.

no I'm not! I'm sorry if I seemed like I was. I just don't like being hanged by a thread. 
He _did_ react in 12 hours but then went silent for 36. I thought he just forgot about me or something so I thought maybe I should ask someone else by opening an issue. And lo and behold he's still here, he just didn't respond (which was fine).
> The statement "no useful answers in 2 days", when derselbst reacted to your question 12 hours after you posted it is quite offensive

I wasn't lying though, I just told it how it was. Although I guess it **was** a bit rude, and I could've been more polite. sorry about that I guess.
Anyways, I hope you have a fine day. I guess I can't get away without learning more C. Cheers.

**Comment by mawe42** [comment]
For the future: please, please don't ever write sentences like "if you don't want to keep maintaining this open source project just stop man" ever again, in any project! Just don't.
Open-source project maintainers are a rare breed, good and active maintainers like derselbst even more so. Suggesting to him to drop the ball is not only stupid (because it doesn't help you in any way), it's also extremely selfish. If he would actually take your advice (maybe because he's fed up with being pressured and harassed for doing all this work), it would leave the FluidSynth community without an active maintainer. Or are you suggesting that you want the job? I very much doubt that.

> I guess I can't get away without learning more C.

That's always a good idea when trying to program against a C API. And what you could also do: continue the discussion in original thread. Maybe start with a "Thank you for trying to help me solve my problem". Then maybe ask if there is any other information that you can provide that might be useful. Maybe give a proper sorry for pressuring you (without "I guess").

Keep the questions coming, be polite, maybe even at some point help out others. Welcome to the FluidSynth community.

**Comment by xXDarkCubeXx** [comment]
> Maybe give a proper sorry for pressuring you (without "I guess").

wdym? the "I guess" isn't supposed to take away from the apology, it's just a linguistic artefact that's supposed to be translated to "I admit I was wrong there, so I apologize" at lease that's how I see it.

And wdym "pressuring him"? Again, I opened this issue because I gave up on hearing back from him.
Trust me man, I'm trying the best I could to sound as polite as possible. But the dude just doesn't want to help me.
